examples/01-filter/image-fft.py

```diff
@@ -0,0 +1,94 @@
+"""
+.. _image_fft_example:
+
+Fast Fourier Transform
+~~~~~~~~~~~~~~~~~~~~~~
+
+This example shows how to apply a Fast Fourier Transform (FFT) to a
+:class:`pyvista.UniformGrid` using :func:`pyvista.UniformGridFilters.fft`
+filter.
+
+Here, we demonstrate FFT usage by denoising an image, effectively removing any
+"high frequency" content by effectively performing a `low pass filter
+<https://en.wikipedia.org/wiki/Low-pass_filter>`_
+
+This example was inspired by `Image denoising by FFT
+<https://scipy-lectures.org/intro/scipy/auto_examples/solutions/plot_fft_image_denoise.html>`_.
+
+"""
+
+import pyvista as pv
+from pyvista import examples
+
+###############################################################################
+# Load the example image moon-landing image and plot it.
+
+image = examples.download_moonlanding_image()
+print(image.point_data)
+
+# Create a theme that we can reuse when plotting the image
+grey_theme = pv.themes.DocumentTheme()
+grey_theme.cmap = 'gray'
+grey_theme.show_scalar_bar = False
+grey_theme.axes.show = False
+image.plot(theme=grey_theme, cpos='xy', text='Unprocessed Moon Landing Image')
+
+###############################################################################
+# Apply FFT to the image
+#
+# FFT will be applied to the active scalars, which is stored in ``'PNGImage'``.
+# The output from the filter contains both real and imaginary components and is
+# stored in the same array.
+
+fft_image = image.fft()
+fft_image.point_data
+
+###############################################################################
+# Plot the FFT of the image. Note that we are effectively viewing the
+# "frequency" of the data in this image, where the four corners contain the low
+# frequency content of the image, and the middle is the high frequency content
+# of the image.
+#
+# .. note::
+#    VTK internally creates a normalized array to plot both the real and
+#    imaginary values from the FFT filter. To avoid having this array included
+#    in the dataset, we use :func:`copy() <DataObject.copy>` to create a
+#    temporary copy that's plotted.
```

Comment:
This is already an existing problem:

```py
Create a cube and clear its point data.

>>> import pyvista as pv
>>> import numpy as np
>>> cube = pv.Cube()
>>> cube.clear_data()
>>> cube.point_data
pyvista DataSetAttributes
Association     : POINT
Active Scalars  : None
Active Vectors  : None
Active Texture  : None
Active Normals  : None
Contains arrays : None

Plot it. Show that by merely plotting we modify the data.

>>> data = np.random.random(cube.n_points)
>>> cube.plot(scalars=data, off_screen=True)
>>> cube.point_data
pyvista DataSetAttributes
Association     : POINT
Active Scalars  : Data-normed
Active Vectors  : None
Active Texture  : None
Active Normals  : None
Contains arrays :
    Data                    float64    (8,)                 SCALARS

Same thing happens with multi-component data.

>>> cube.clear_data()
>>> data = np.random.random((cube.n_points, 3))
>>> cube.plot(scalars=data, off_screen=True)
>>> cube.point_data
pyvista DataSetAttributes
Association     : POINT
Active Scalars  : Data-normed
Active Vectors  : None
Active Texture  : None
Active Normals  : None
Contains arrays :
    Data-normed             float64    (8,)                 SCALARS
```

So... we either fix the current problem in a different PR or we leave it alone.

Usually with these examples we uncover 15 issues...

Reply:
I see, thanks for the explanation. The problem is that the array is added in `Plotter.add_mesh()` (etc.), and removing the array before `pl.show()` removes the scalars from plotting too (semi-obviously). So I'm not sure we could do much... we could clear the array at the end of `plot()`, but that will still leave manual `Plotter` use in the original state, and it might interfere with notebook plotting.

So yeah, definitely don't do anything with this here, and maybe do something with this in a separate PR, but I'm not sure what we could do.

Reply:
This will be resolved by #2763. Once that's merged I'll fix it here.

Reply:
Fixed in 421c833da6c6e1b71a913fab5c086a2198b9c932.